api/v1alpha1/kepler_x_types.go

```diff
@@ -0,0 +1,176 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NOTE: all internal types can depend on public types
+
+type KeplerXDeploymentSpec struct {
+	// +kubebuilder:validation:MinLength=3
+	Image string `json:"image"`
+
+	// +kubebuilder:validation:MinLength=1
+	Namespace string `json:"namespace"`
+
+	// +kubebuilder:default=9103
+	// +kubebuilder:validation:Maximum=65535
+	// +kubebuilder:validation:Minimum=1
+	Port int32 `json:"port,omitempty"`
+
+	// Defines which Nodes the Pod is scheduled on
+	// +optional
+	// +kubebuilder:default={"kubernetes.io/os":"linux"}
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// If specified, define Pod's tolerations
+	// +optional
+	// +kubebuilder:default={{"key": "", "operator": "Exists", "value": "", "effect": ""}}
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+}
+
+type KeplerXConfigSpec struct {
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Maximum=500
+	// +kubebuilder:validation:Minimum=1
+	MonitorInterval int32 `json:"monitorInterval,omitempty"`
+
+	// +kubebuilder:default=3
+	// +kubebuilder:validation:Maximum=500
+	// +kubebuilder:validation:Minimum=1
+	ExporterInterval int32 `json:"exporterInterval,omitempty"`
```

Comment:
We don't need this .. and its pretty hard to specify intervals of all prometheus scrapping the exporter. It should function without that knowledge.